elixir/apps/api/lib/api/client/channel.ex

```diff
@@ -507,6 +552,90 @@ defmodule API.Client.Channel do
+  #
+  # `connected_gateway_ids` is used to indicate that the client is already connected to some of the gateways,
+  # so the gateway can be reused by multiplexing the connection.
+  #
+  # `client_payload` is an arbitrary data that the client wants to send to the gateway, typically it will be
+  # either empty, or contain dns hostname of a given resource to connect to (so that it can be resolved on the
+  # gateway).
+  def handle_in(
+        "create_flow",
+        %{
+          "resource_id" => resource_id,
+          "connected_gateway_ids" => connected_gateway_ids
+        } = attrs,
+        socket
+      ) do
+    OpenTelemetry.Ctx.attach(socket.assigns.opentelemetry_ctx)
+    OpenTelemetry.Tracer.set_current_span(socket.assigns.opentelemetry_span_ctx)
+
+    OpenTelemetry.Tracer.with_span "client.create_flow", attributes: attrs do
+      location = {
+        socket.assigns.client.last_seen_remote_ip_location_lat,
+        socket.assigns.client.last_seen_remote_ip_location_lon
+      }
+
+      with {:ok, resource} <-
+             Resources.fetch_and_authorize_resource_by_id(resource_id, socket.assigns.subject),
+           {:ok, gateways} when gateways != [] <-
+             Gateways.all_connected_gateways_for_resource(resource, socket.assigns.subject,
+               preload: :group
+             ),
+           {:ok, gateways} <-
+             filter_compatible_gateways(gateways, socket.assigns.gateway_version_requirement),
+           OpenTelemetry.Tracer.set_attribute(:gateways_count, length(gateways)),
+           gateway = Gateways.load_balance_gateways(location, gateways, connected_gateway_ids),
+           OpenTelemetry.Tracer.set_attribute(:gateway_id, gateway.id),
+           {:ok, resource, flow} <-
+             Flows.authorize_flow(
+               socket.assigns.client,
+               gateway,
+               resource_id,
+               socket.assigns.subject
+             ) do
+        OpenTelemetry.Tracer.set_attribute(:flow_id, flow.id)
+        opentelemetry_ctx = OpenTelemetry.Ctx.get_current()
+        opentelemetry_span_ctx = OpenTelemetry.Tracer.current_span_ctx()
+
+        preshared_key = Domain.Crypto.psk()
+        ice_credentials = generate_ice_credentials(socket.assigns.client, gateway)
+
+        :ok =
+          Gateways.broadcast_to_gateway(
+            gateway,
+            {:authorize_flow, {self(), socket_ref(socket)},
+             %{
+               client_id: socket.assigns.client.id,
+               resource_id: resource.id,
+               flow_id: flow.id,
+               authorization_expires_at: flow.expires_at,
+               ice_credentials: ice_credentials,
+               preshared_key: preshared_key
+             }, {opentelemetry_ctx, opentelemetry_span_ctx}}
+          )
+
+        {:noreply, socket}
+      else
+        {:error, :not_found} ->
+          OpenTelemetry.Tracer.set_status(:error, "not_found")
+          {:reply, {:error, %{reason: :not_found}}, socket}
```

Comment:
If we decide to use a new message for the success case, it would be good to have a dedicated message for failures too. They'd have to carry the resource ID as contextual information.

Something like:

- `create_flow_ok`
- `create_flow_err`